goalc/compiler/compilation/Macro.cpp

```diff
@@ -203,7 +206,8 @@ Val* Compiler::compile_define_constant(const goos::Object& form,
     m_goos.global_environment.as_env()->vars[sym] = value;
   }
 
-  m_symbol_info.add_constant(sym->name, form);
+  m_global_constant_types.emplace(sym->name, value);
```

Comment:
It looks like this isn't really types, but just the value of the constant. I'd recommend just using `m_gloal_constants` instead.